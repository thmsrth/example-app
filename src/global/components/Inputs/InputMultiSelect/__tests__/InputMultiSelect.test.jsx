import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import InputMultiSelect from '../InputMultiSelect';

const props = {
  className: 'MyInputMultiSelect',
  name: 'MyInputName',
  label: 'My Input Label',
  options: [
    {
      value: 'first_value',
      label: 'First Value',
    },
    {
      value: 'second_value',
      label: 'Second Value',
    },
  ],
  value: ['first_value'],
  disabled: false,
  focused: false,
  validationKeys: {
    required: false,
    validationAlert: '',
    warningAlert: '',
  },
  handleChange: () => {},
  handleFocus: () => {},
  handleBlur: () => {},
};

describe('InputMultiSelect', () => {
  it('should render the component with first value', () => {
    const tree = shallow(<InputMultiSelect {...props} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should render the component with nothing selected', () => {
    const tree = shallow(<InputMultiSelect {...props} value={[]} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should render the component with second value', () => {
    const tree = shallow(<InputMultiSelect {...props} value={['second_value']} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should render the component with focused classes while focused', () => {
    const tree = shallow(<InputMultiSelect {...props} focused />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should render the component with disabled classes while disabled', () => {
    const tree = shallow(<InputMultiSelect {...props} disabled />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should render the component as a required input', () => {
    const tree = shallow(<InputMultiSelect
      {...props}
      validationKeys={{ required: true }}
    />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should render the component as a required input', () => {
    const tree = shallow(<InputMultiSelect
      {...props}
      validationKeys={{ required: true }}
    />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should render the component with a validation alert message', () => {
    const tree = shallow(<InputMultiSelect
      {...props}
      validationKeys={{ validationAlert: 'My validation alert message' }}
    />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should render the component with a warning message', () => {
    const tree = shallow(<InputMultiSelect
      {...props}
      validationKeys={{ warningAlert: 'My warning message' }}
    />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should render the component with is loading text', () => {
    const tree = shallow(<InputMultiSelect {...props} value={[]} loading="Is loading ..." />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should render the component with is error message', () => {
    const tree = shallow(<InputMultiSelect {...props} value={[]} error="Error message" />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
