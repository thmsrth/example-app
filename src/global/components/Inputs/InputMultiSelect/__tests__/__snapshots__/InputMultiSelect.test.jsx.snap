// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputMultiSelect should render the component as a required input 1`] = `
<div
  className="InputMultiSelect MyInputMultiSelect InputMultiSelect__MyInputName MyInputMultiSelect__MyInputName InputMultiSelect__required MyInputMultiSelect__required InputMultiSelect--alert MyInputMultiSelect--alert"
>
  <label
    className="InputMultiSelect__label MyInputMultiSelect__label MyInputMultiSelect__MyInputName__label "
    key="MyInputName"
  >
    My Input Label*
    <div
      className="InputMultiSelect__alertMsg MyInputMultiSelect__alertMsg InputMultiSelect__MyInputName__alertMsg MyInputMultiSelect__MyInputName__alertMsg"
    />
    <div
      className="InputMultiSelect__input MyInputMultiSelect__input  "
    >
      <Select
        arrowRenderer={[Function]}
        autosize={true}
        backspaceRemoves={true}
        backspaceToRemoveMessage="Press backspace to remove {label}"
        className="InputMultiSelect__select MyInputMultiSelect__select InputMultiSelect__select--alert MyInputMultiSelect__select--alert"
        clearAllText="Clear all"
        clearRenderer={[Function]}
        clearValueText="Clear value"
        clearable={false}
        closeOnSelect={true}
        deleteRemoves={true}
        delimiter=","
        disabled={false}
        escapeClearsValue={true}
        filterOptions={[Function]}
        ignoreAccents={true}
        ignoreCase={true}
        inputProps={Object {}}
        isLoading={false}
        joinValues={false}
        labelKey="label"
        matchPos="any"
        matchProp="any"
        menuBuffer={0}
        menuRenderer={[Function]}
        multi={true}
        name="MyInputName"
        noResultsText="No results found"
        onBlur={[Function]}
        onBlurResetsInput={true}
        onChange={[Function]}
        onCloseResetsInput={true}
        onFocus={[Function]}
        onSelectResetsInput={true}
        openOnClick={true}
        optionComponent={[Function]}
        options={
          Array [
            Object {
              "label": "First Value",
              "value": "first_value",
            },
            Object {
              "label": "Second Value",
              "value": "second_value",
            },
          ]
        }
        pageSize={5}
        placeholder="Nothing selected"
        removeSelected={true}
        required={true}
        rtl={false}
        scrollMenuIntoView={true}
        searchable={false}
        simpleValue={false}
        tabSelectsValue={true}
        trimFilter={true}
        value={
          Array [
            "first_value",
          ]
        }
        valueComponent={[Function]}
        valueKey="value"
      />
    </div>
    <div
      className="InputMultiSelect__warningMsg MyInputMultiSelect__warningMsg InputMultiSelect__MyInputName__warningMsg MyInputMultiSelect__MyInputName__warningMsg"
    />
  </label>
</div>
`;

exports[`InputMultiSelect should render the component as a required input 2`] = `
<div
  className="InputMultiSelect MyInputMultiSelect InputMultiSelect__MyInputName MyInputMultiSelect__MyInputName InputMultiSelect__required MyInputMultiSelect__required InputMultiSelect--alert MyInputMultiSelect--alert"
>
  <label
    className="InputMultiSelect__label MyInputMultiSelect__label MyInputMultiSelect__MyInputName__label "
    key="MyInputName"
  >
    My Input Label*
    <div
      className="InputMultiSelect__alertMsg MyInputMultiSelect__alertMsg InputMultiSelect__MyInputName__alertMsg MyInputMultiSelect__MyInputName__alertMsg"
    />
    <div
      className="InputMultiSelect__input MyInputMultiSelect__input  "
    >
      <Select
        arrowRenderer={[Function]}
        autosize={true}
        backspaceRemoves={true}
        backspaceToRemoveMessage="Press backspace to remove {label}"
        className="InputMultiSelect__select MyInputMultiSelect__select InputMultiSelect__select--alert MyInputMultiSelect__select--alert"
        clearAllText="Clear all"
        clearRenderer={[Function]}
        clearValueText="Clear value"
        clearable={false}
        closeOnSelect={true}
        deleteRemoves={true}
        delimiter=","
        disabled={false}
        escapeClearsValue={true}
        filterOptions={[Function]}
        ignoreAccents={true}
        ignoreCase={true}
        inputProps={Object {}}
        isLoading={false}
        joinValues={false}
        labelKey="label"
        matchPos="any"
        matchProp="any"
        menuBuffer={0}
        menuRenderer={[Function]}
        multi={true}
        name="MyInputName"
        noResultsText="No results found"
        onBlur={[Function]}
        onBlurResetsInput={true}
        onChange={[Function]}
        onCloseResetsInput={true}
        onFocus={[Function]}
        onSelectResetsInput={true}
        openOnClick={true}
        optionComponent={[Function]}
        options={
          Array [
            Object {
              "label": "First Value",
              "value": "first_value",
            },
            Object {
              "label": "Second Value",
              "value": "second_value",
            },
          ]
        }
        pageSize={5}
        placeholder="Nothing selected"
        removeSelected={true}
        required={true}
        rtl={false}
        scrollMenuIntoView={true}
        searchable={false}
        simpleValue={false}
        tabSelectsValue={true}
        trimFilter={true}
        value={
          Array [
            "first_value",
          ]
        }
        valueComponent={[Function]}
        valueKey="value"
      />
    </div>
    <div
      className="InputMultiSelect__warningMsg MyInputMultiSelect__warningMsg InputMultiSelect__MyInputName__warningMsg MyInputMultiSelect__MyInputName__warningMsg"
    />
  </label>
</div>
`;

exports[`InputMultiSelect should render the component with a validation alert message 1`] = `
<div
  className="InputMultiSelect MyInputMultiSelect InputMultiSelect__MyInputName MyInputMultiSelect__MyInputName InputMultiSelect__required MyInputMultiSelect__required InputMultiSelect--alert MyInputMultiSelect--alert"
>
  <label
    className="InputMultiSelect__label MyInputMultiSelect__label MyInputMultiSelect__MyInputName__label "
    key="MyInputName"
  >
    My Input Label
    <div
      className="InputMultiSelect__alertMsg MyInputMultiSelect__alertMsg InputMultiSelect__MyInputName__alertMsg MyInputMultiSelect__MyInputName__alertMsg"
    >
      My validation alert message
    </div>
    <div
      className="InputMultiSelect__input MyInputMultiSelect__input  "
    >
      <Select
        arrowRenderer={[Function]}
        autosize={true}
        backspaceRemoves={true}
        backspaceToRemoveMessage="Press backspace to remove {label}"
        className="InputMultiSelect__select MyInputMultiSelect__select InputMultiSelect__select--alert MyInputMultiSelect__select--alert"
        clearAllText="Clear all"
        clearRenderer={[Function]}
        clearValueText="Clear value"
        clearable={false}
        closeOnSelect={true}
        deleteRemoves={true}
        delimiter=","
        disabled={false}
        escapeClearsValue={true}
        filterOptions={[Function]}
        ignoreAccents={true}
        ignoreCase={true}
        inputProps={Object {}}
        isLoading={false}
        joinValues={false}
        labelKey="label"
        matchPos="any"
        matchProp="any"
        menuBuffer={0}
        menuRenderer={[Function]}
        multi={true}
        name="MyInputName"
        noResultsText="No results found"
        onBlur={[Function]}
        onBlurResetsInput={true}
        onChange={[Function]}
        onCloseResetsInput={true}
        onFocus={[Function]}
        onSelectResetsInput={true}
        openOnClick={true}
        optionComponent={[Function]}
        options={
          Array [
            Object {
              "label": "First Value",
              "value": "first_value",
            },
            Object {
              "label": "Second Value",
              "value": "second_value",
            },
          ]
        }
        pageSize={5}
        placeholder="Nothing selected"
        removeSelected={true}
        required={false}
        rtl={false}
        scrollMenuIntoView={true}
        searchable={false}
        simpleValue={false}
        tabSelectsValue={true}
        trimFilter={true}
        value={
          Array [
            "first_value",
          ]
        }
        valueComponent={[Function]}
        valueKey="value"
      />
    </div>
    <div
      className="InputMultiSelect__warningMsg MyInputMultiSelect__warningMsg InputMultiSelect__MyInputName__warningMsg MyInputMultiSelect__MyInputName__warningMsg"
    />
  </label>
</div>
`;

exports[`InputMultiSelect should render the component with a warning message 1`] = `
<div
  className="InputMultiSelect MyInputMultiSelect InputMultiSelect__MyInputName MyInputMultiSelect__MyInputName InputMultiSelect__required MyInputMultiSelect__required InputMultiSelect--alert MyInputMultiSelect--alert"
>
  <label
    className="InputMultiSelect__label MyInputMultiSelect__label MyInputMultiSelect__MyInputName__label "
    key="MyInputName"
  >
    My Input Label
    <div
      className="InputMultiSelect__alertMsg MyInputMultiSelect__alertMsg InputMultiSelect__MyInputName__alertMsg MyInputMultiSelect__MyInputName__alertMsg"
    />
    <div
      className="InputMultiSelect__input MyInputMultiSelect__input  "
    >
      <Select
        arrowRenderer={[Function]}
        autosize={true}
        backspaceRemoves={true}
        backspaceToRemoveMessage="Press backspace to remove {label}"
        className="InputMultiSelect__select MyInputMultiSelect__select InputMultiSelect__select--alert MyInputMultiSelect__select--alert"
        clearAllText="Clear all"
        clearRenderer={[Function]}
        clearValueText="Clear value"
        clearable={false}
        closeOnSelect={true}
        deleteRemoves={true}
        delimiter=","
        disabled={false}
        escapeClearsValue={true}
        filterOptions={[Function]}
        ignoreAccents={true}
        ignoreCase={true}
        inputProps={Object {}}
        isLoading={false}
        joinValues={false}
        labelKey="label"
        matchPos="any"
        matchProp="any"
        menuBuffer={0}
        menuRenderer={[Function]}
        multi={true}
        name="MyInputName"
        noResultsText="No results found"
        onBlur={[Function]}
        onBlurResetsInput={true}
        onChange={[Function]}
        onCloseResetsInput={true}
        onFocus={[Function]}
        onSelectResetsInput={true}
        openOnClick={true}
        optionComponent={[Function]}
        options={
          Array [
            Object {
              "label": "First Value",
              "value": "first_value",
            },
            Object {
              "label": "Second Value",
              "value": "second_value",
            },
          ]
        }
        pageSize={5}
        placeholder="Nothing selected"
        removeSelected={true}
        required={false}
        rtl={false}
        scrollMenuIntoView={true}
        searchable={false}
        simpleValue={false}
        tabSelectsValue={true}
        trimFilter={true}
        value={
          Array [
            "first_value",
          ]
        }
        valueComponent={[Function]}
        valueKey="value"
      />
    </div>
    <div
      className="InputMultiSelect__warningMsg MyInputMultiSelect__warningMsg InputMultiSelect__MyInputName__warningMsg MyInputMultiSelect__MyInputName__warningMsg"
    >
      My warning message
    </div>
  </label>
</div>
`;

exports[`InputMultiSelect should render the component with disabled classes while disabled 1`] = `
<div
  className="InputMultiSelect MyInputMultiSelect InputMultiSelect__MyInputName MyInputMultiSelect__MyInputName  "
>
  <label
    className="InputMultiSelect__label MyInputMultiSelect__label MyInputMultiSelect__MyInputName__label InputMultiSelect__label--disabled MyInputMultiSelect__label--disabled MyInputMultiSelect__MyInputName__label--disabled"
    key="MyInputName"
  >
    My Input Label
    <div
      className="InputMultiSelect__alertMsg MyInputMultiSelect__alertMsg InputMultiSelect__MyInputName__alertMsg MyInputMultiSelect__MyInputName__alertMsg"
    />
    <div
      className="InputMultiSelect__input MyInputMultiSelect__input  InputMultiSelect__input--disabled MyInputMultiSelect__input--disabled"
    >
      <Select
        arrowRenderer={[Function]}
        autosize={true}
        backspaceRemoves={true}
        backspaceToRemoveMessage="Press backspace to remove {label}"
        className="InputMultiSelect__select MyInputMultiSelect__select "
        clearAllText="Clear all"
        clearRenderer={[Function]}
        clearValueText="Clear value"
        clearable={false}
        closeOnSelect={true}
        deleteRemoves={true}
        delimiter=","
        disabled={true}
        escapeClearsValue={true}
        filterOptions={[Function]}
        ignoreAccents={true}
        ignoreCase={true}
        inputProps={Object {}}
        isLoading={false}
        joinValues={false}
        labelKey="label"
        matchPos="any"
        matchProp="any"
        menuBuffer={0}
        menuRenderer={[Function]}
        multi={true}
        name="MyInputName"
        noResultsText="No results found"
        onBlur={[Function]}
        onBlurResetsInput={true}
        onChange={[Function]}
        onCloseResetsInput={true}
        onFocus={[Function]}
        onSelectResetsInput={true}
        openOnClick={true}
        optionComponent={[Function]}
        options={
          Array [
            Object {
              "label": "First Value",
              "value": "first_value",
            },
            Object {
              "label": "Second Value",
              "value": "second_value",
            },
          ]
        }
        pageSize={5}
        placeholder="Nothing selected"
        removeSelected={true}
        required={false}
        rtl={false}
        scrollMenuIntoView={true}
        searchable={false}
        simpleValue={false}
        tabSelectsValue={true}
        trimFilter={true}
        value={
          Array [
            "first_value",
          ]
        }
        valueComponent={[Function]}
        valueKey="value"
      />
    </div>
    <div
      className="InputMultiSelect__warningMsg MyInputMultiSelect__warningMsg InputMultiSelect__MyInputName__warningMsg MyInputMultiSelect__MyInputName__warningMsg"
    />
  </label>
</div>
`;

exports[`InputMultiSelect should render the component with first value 1`] = `
<div
  className="InputMultiSelect MyInputMultiSelect InputMultiSelect__MyInputName MyInputMultiSelect__MyInputName  "
>
  <label
    className="InputMultiSelect__label MyInputMultiSelect__label MyInputMultiSelect__MyInputName__label "
    key="MyInputName"
  >
    My Input Label
    <div
      className="InputMultiSelect__alertMsg MyInputMultiSelect__alertMsg InputMultiSelect__MyInputName__alertMsg MyInputMultiSelect__MyInputName__alertMsg"
    />
    <div
      className="InputMultiSelect__input MyInputMultiSelect__input  "
    >
      <Select
        arrowRenderer={[Function]}
        autosize={true}
        backspaceRemoves={true}
        backspaceToRemoveMessage="Press backspace to remove {label}"
        className="InputMultiSelect__select MyInputMultiSelect__select "
        clearAllText="Clear all"
        clearRenderer={[Function]}
        clearValueText="Clear value"
        clearable={false}
        closeOnSelect={true}
        deleteRemoves={true}
        delimiter=","
        disabled={false}
        escapeClearsValue={true}
        filterOptions={[Function]}
        ignoreAccents={true}
        ignoreCase={true}
        inputProps={Object {}}
        isLoading={false}
        joinValues={false}
        labelKey="label"
        matchPos="any"
        matchProp="any"
        menuBuffer={0}
        menuRenderer={[Function]}
        multi={true}
        name="MyInputName"
        noResultsText="No results found"
        onBlur={[Function]}
        onBlurResetsInput={true}
        onChange={[Function]}
        onCloseResetsInput={true}
        onFocus={[Function]}
        onSelectResetsInput={true}
        openOnClick={true}
        optionComponent={[Function]}
        options={
          Array [
            Object {
              "label": "First Value",
              "value": "first_value",
            },
            Object {
              "label": "Second Value",
              "value": "second_value",
            },
          ]
        }
        pageSize={5}
        placeholder="Nothing selected"
        removeSelected={true}
        required={false}
        rtl={false}
        scrollMenuIntoView={true}
        searchable={false}
        simpleValue={false}
        tabSelectsValue={true}
        trimFilter={true}
        value={
          Array [
            "first_value",
          ]
        }
        valueComponent={[Function]}
        valueKey="value"
      />
    </div>
    <div
      className="InputMultiSelect__warningMsg MyInputMultiSelect__warningMsg InputMultiSelect__MyInputName__warningMsg MyInputMultiSelect__MyInputName__warningMsg"
    />
  </label>
</div>
`;

exports[`InputMultiSelect should render the component with focused classes while focused 1`] = `
<div
  className="InputMultiSelect MyInputMultiSelect InputMultiSelect__MyInputName MyInputMultiSelect__MyInputName  "
>
  <label
    className="InputMultiSelect__label MyInputMultiSelect__label MyInputMultiSelect__MyInputName__label "
    key="MyInputName"
  >
    My Input Label
    <div
      className="InputMultiSelect__alertMsg MyInputMultiSelect__alertMsg InputMultiSelect__MyInputName__alertMsg MyInputMultiSelect__MyInputName__alertMsg"
    />
    <div
      className="InputMultiSelect__input MyInputMultiSelect__input InputMultiSelect__input--focused MyInputMultiSelect__input--focused "
    >
      <Select
        arrowRenderer={[Function]}
        autosize={true}
        backspaceRemoves={true}
        backspaceToRemoveMessage="Press backspace to remove {label}"
        className="InputMultiSelect__select MyInputMultiSelect__select "
        clearAllText="Clear all"
        clearRenderer={[Function]}
        clearValueText="Clear value"
        clearable={false}
        closeOnSelect={true}
        deleteRemoves={true}
        delimiter=","
        disabled={false}
        escapeClearsValue={true}
        filterOptions={[Function]}
        ignoreAccents={true}
        ignoreCase={true}
        inputProps={Object {}}
        isLoading={false}
        joinValues={false}
        labelKey="label"
        matchPos="any"
        matchProp="any"
        menuBuffer={0}
        menuRenderer={[Function]}
        multi={true}
        name="MyInputName"
        noResultsText="No results found"
        onBlur={[Function]}
        onBlurResetsInput={true}
        onChange={[Function]}
        onCloseResetsInput={true}
        onFocus={[Function]}
        onSelectResetsInput={true}
        openOnClick={true}
        optionComponent={[Function]}
        options={
          Array [
            Object {
              "label": "First Value",
              "value": "first_value",
            },
            Object {
              "label": "Second Value",
              "value": "second_value",
            },
          ]
        }
        pageSize={5}
        placeholder="Nothing selected"
        removeSelected={true}
        required={false}
        rtl={false}
        scrollMenuIntoView={true}
        searchable={false}
        simpleValue={false}
        tabSelectsValue={true}
        trimFilter={true}
        value={
          Array [
            "first_value",
          ]
        }
        valueComponent={[Function]}
        valueKey="value"
      />
    </div>
    <div
      className="InputMultiSelect__warningMsg MyInputMultiSelect__warningMsg InputMultiSelect__MyInputName__warningMsg MyInputMultiSelect__MyInputName__warningMsg"
    />
  </label>
</div>
`;

exports[`InputMultiSelect should render the component with is error message 1`] = `
<div
  className="InputMultiSelect MyInputMultiSelect InputMultiSelect__MyInputName MyInputMultiSelect__MyInputName  InputMultiSelect--alert MyInputMultiSelect--alert"
>
  <label
    className="InputMultiSelect__label MyInputMultiSelect__label MyInputMultiSelect__MyInputName__label "
    key="MyInputName"
  >
    My Input Label
    <div
      className="InputMultiSelect__alertMsg MyInputMultiSelect__alertMsg InputMultiSelect__MyInputName__alertMsg MyInputMultiSelect__MyInputName__alertMsg"
    >
      Error message
    </div>
    <div
      className="InputMultiSelect__input MyInputMultiSelect__input  "
    >
      <Select
        arrowRenderer={[Function]}
        autosize={true}
        backspaceRemoves={true}
        backspaceToRemoveMessage="Press backspace to remove {label}"
        className="InputMultiSelect__select MyInputMultiSelect__select InputMultiSelect__select--alert MyInputMultiSelect__select--alert"
        clearAllText="Clear all"
        clearRenderer={[Function]}
        clearValueText="Clear value"
        clearable={false}
        closeOnSelect={true}
        deleteRemoves={true}
        delimiter=","
        disabled={false}
        escapeClearsValue={true}
        filterOptions={[Function]}
        ignoreAccents={true}
        ignoreCase={true}
        inputProps={Object {}}
        isLoading={false}
        joinValues={false}
        labelKey="label"
        matchPos="any"
        matchProp="any"
        menuBuffer={0}
        menuRenderer={[Function]}
        multi={true}
        name="MyInputName"
        noResultsText="No results found"
        onBlur={[Function]}
        onBlurResetsInput={true}
        onChange={[Function]}
        onCloseResetsInput={true}
        onFocus={[Function]}
        onSelectResetsInput={true}
        openOnClick={true}
        optionComponent={[Function]}
        options={
          Array [
            Object {
              "label": "First Value",
              "value": "first_value",
            },
            Object {
              "label": "Second Value",
              "value": "second_value",
            },
          ]
        }
        pageSize={5}
        placeholder="Nothing selected"
        removeSelected={true}
        required={false}
        rtl={false}
        scrollMenuIntoView={true}
        searchable={false}
        simpleValue={false}
        tabSelectsValue={true}
        trimFilter={true}
        value={Array []}
        valueComponent={[Function]}
        valueKey="value"
      />
    </div>
    <div
      className="InputMultiSelect__warningMsg MyInputMultiSelect__warningMsg InputMultiSelect__MyInputName__warningMsg MyInputMultiSelect__MyInputName__warningMsg"
    />
  </label>
</div>
`;

exports[`InputMultiSelect should render the component with is loading text 1`] = `
<div
  className="InputMultiSelect MyInputMultiSelect InputMultiSelect__MyInputName MyInputMultiSelect__MyInputName  "
>
  <label
    className="InputMultiSelect__label MyInputMultiSelect__label MyInputMultiSelect__MyInputName__label "
    key="MyInputName"
  >
    My Input Label
    <div
      className="InputMultiSelect__alertMsg MyInputMultiSelect__alertMsg InputMultiSelect__MyInputName__alertMsg MyInputMultiSelect__MyInputName__alertMsg"
    />
    <div
      className="InputMultiSelect__input MyInputMultiSelect__input  "
    >
      <Select
        arrowRenderer={[Function]}
        autosize={true}
        backspaceRemoves={true}
        backspaceToRemoveMessage="Press backspace to remove {label}"
        className="InputMultiSelect__select MyInputMultiSelect__select "
        clearAllText="Clear all"
        clearRenderer={[Function]}
        clearValueText="Clear value"
        clearable={false}
        closeOnSelect={true}
        deleteRemoves={true}
        delimiter=","
        disabled={false}
        escapeClearsValue={true}
        filterOptions={[Function]}
        ignoreAccents={true}
        ignoreCase={true}
        inputProps={Object {}}
        isLoading={true}
        joinValues={false}
        labelKey="label"
        matchPos="any"
        matchProp="any"
        menuBuffer={0}
        menuRenderer={[Function]}
        multi={true}
        name="MyInputName"
        noResultsText="No results found"
        onBlur={[Function]}
        onBlurResetsInput={true}
        onChange={[Function]}
        onCloseResetsInput={true}
        onFocus={[Function]}
        onSelectResetsInput={true}
        openOnClick={true}
        optionComponent={[Function]}
        options={
          Array [
            Object {
              "label": "First Value",
              "value": "first_value",
            },
            Object {
              "label": "Second Value",
              "value": "second_value",
            },
          ]
        }
        pageSize={5}
        placeholder="Is loading ..."
        removeSelected={true}
        required={false}
        rtl={false}
        scrollMenuIntoView={true}
        searchable={false}
        simpleValue={false}
        tabSelectsValue={true}
        trimFilter={true}
        value={Array []}
        valueComponent={[Function]}
        valueKey="value"
      />
    </div>
    <div
      className="InputMultiSelect__warningMsg MyInputMultiSelect__warningMsg InputMultiSelect__MyInputName__warningMsg MyInputMultiSelect__MyInputName__warningMsg"
    />
  </label>
</div>
`;

exports[`InputMultiSelect should render the component with nothing selected 1`] = `
<div
  className="InputMultiSelect MyInputMultiSelect InputMultiSelect__MyInputName MyInputMultiSelect__MyInputName  "
>
  <label
    className="InputMultiSelect__label MyInputMultiSelect__label MyInputMultiSelect__MyInputName__label "
    key="MyInputName"
  >
    My Input Label
    <div
      className="InputMultiSelect__alertMsg MyInputMultiSelect__alertMsg InputMultiSelect__MyInputName__alertMsg MyInputMultiSelect__MyInputName__alertMsg"
    />
    <div
      className="InputMultiSelect__input MyInputMultiSelect__input  "
    >
      <Select
        arrowRenderer={[Function]}
        autosize={true}
        backspaceRemoves={true}
        backspaceToRemoveMessage="Press backspace to remove {label}"
        className="InputMultiSelect__select MyInputMultiSelect__select "
        clearAllText="Clear all"
        clearRenderer={[Function]}
        clearValueText="Clear value"
        clearable={false}
        closeOnSelect={true}
        deleteRemoves={true}
        delimiter=","
        disabled={false}
        escapeClearsValue={true}
        filterOptions={[Function]}
        ignoreAccents={true}
        ignoreCase={true}
        inputProps={Object {}}
        isLoading={false}
        joinValues={false}
        labelKey="label"
        matchPos="any"
        matchProp="any"
        menuBuffer={0}
        menuRenderer={[Function]}
        multi={true}
        name="MyInputName"
        noResultsText="No results found"
        onBlur={[Function]}
        onBlurResetsInput={true}
        onChange={[Function]}
        onCloseResetsInput={true}
        onFocus={[Function]}
        onSelectResetsInput={true}
        openOnClick={true}
        optionComponent={[Function]}
        options={
          Array [
            Object {
              "label": "First Value",
              "value": "first_value",
            },
            Object {
              "label": "Second Value",
              "value": "second_value",
            },
          ]
        }
        pageSize={5}
        placeholder="Nothing selected"
        removeSelected={true}
        required={false}
        rtl={false}
        scrollMenuIntoView={true}
        searchable={false}
        simpleValue={false}
        tabSelectsValue={true}
        trimFilter={true}
        value={Array []}
        valueComponent={[Function]}
        valueKey="value"
      />
    </div>
    <div
      className="InputMultiSelect__warningMsg MyInputMultiSelect__warningMsg InputMultiSelect__MyInputName__warningMsg MyInputMultiSelect__MyInputName__warningMsg"
    />
  </label>
</div>
`;

exports[`InputMultiSelect should render the component with second value 1`] = `
<div
  className="InputMultiSelect MyInputMultiSelect InputMultiSelect__MyInputName MyInputMultiSelect__MyInputName  "
>
  <label
    className="InputMultiSelect__label MyInputMultiSelect__label MyInputMultiSelect__MyInputName__label "
    key="MyInputName"
  >
    My Input Label
    <div
      className="InputMultiSelect__alertMsg MyInputMultiSelect__alertMsg InputMultiSelect__MyInputName__alertMsg MyInputMultiSelect__MyInputName__alertMsg"
    />
    <div
      className="InputMultiSelect__input MyInputMultiSelect__input  "
    >
      <Select
        arrowRenderer={[Function]}
        autosize={true}
        backspaceRemoves={true}
        backspaceToRemoveMessage="Press backspace to remove {label}"
        className="InputMultiSelect__select MyInputMultiSelect__select "
        clearAllText="Clear all"
        clearRenderer={[Function]}
        clearValueText="Clear value"
        clearable={false}
        closeOnSelect={true}
        deleteRemoves={true}
        delimiter=","
        disabled={false}
        escapeClearsValue={true}
        filterOptions={[Function]}
        ignoreAccents={true}
        ignoreCase={true}
        inputProps={Object {}}
        isLoading={false}
        joinValues={false}
        labelKey="label"
        matchPos="any"
        matchProp="any"
        menuBuffer={0}
        menuRenderer={[Function]}
        multi={true}
        name="MyInputName"
        noResultsText="No results found"
        onBlur={[Function]}
        onBlurResetsInput={true}
        onChange={[Function]}
        onCloseResetsInput={true}
        onFocus={[Function]}
        onSelectResetsInput={true}
        openOnClick={true}
        optionComponent={[Function]}
        options={
          Array [
            Object {
              "label": "First Value",
              "value": "first_value",
            },
            Object {
              "label": "Second Value",
              "value": "second_value",
            },
          ]
        }
        pageSize={5}
        placeholder="Nothing selected"
        removeSelected={true}
        required={false}
        rtl={false}
        scrollMenuIntoView={true}
        searchable={false}
        simpleValue={false}
        tabSelectsValue={true}
        trimFilter={true}
        value={
          Array [
            "second_value",
          ]
        }
        valueComponent={[Function]}
        valueKey="value"
      />
    </div>
    <div
      className="InputMultiSelect__warningMsg MyInputMultiSelect__warningMsg InputMultiSelect__MyInputName__warningMsg MyInputMultiSelect__MyInputName__warningMsg"
    />
  </label>
</div>
`;
